### ISSUE TITLE
Interface order and minimum required PHP version

### ISSUE THREAD

**Issue by ganiuszka** [pradosoft/prado#809]
Hello the PRADO Group,

First I would say big thank you for the release 4.2.0. It is a huge amount of work and effort to provide this new version. I really appreciate it.

I have a question to the release notes for 4.2.0 and the minimum required PHP version (7.3). I tried it with PHP 7.3 and with this PHP there happens the following error:

```
[Compile Error] Class Prado\Web\UI\ActiveControls\TActiveDropDownList cannot implement previously implemented interface Prado\Web\UI\ActiveControls\IActiveControl (@line 37 in file /var/www/bacularis-app/protected/vendor/pradosoft/prado/framework/Web/UI/ActiveControls/TActiveDropDownList.php).
```

I did a small research and I am seeing that the problem is in order of using interfaces in ActiveControls. To illustrate it, I will use two code samples where one is correct, and one finishes with the described error:

Correct:
```
interface A {
}

interface B extends A {
}

class C implements A, B {
}
```

Causes error:
```
interface A {
}

interface B extends A {
}

class C implements B, A {
}
```

The same situation is with `ICallbackEventHandler` and `IActiveControl` interfaces. The first one extends the second one and both of them are used in active controls in the order that causes this error.

```
class TActiveDropDownList extends TDropDownList implements ICallbackEventHandler, IActiveControl
```

This change has been made in commit:

63bc9830cdf258384aa445ee62e999c42ad38221

Good news is that this error has been fixed in PHP 7.4 and used interface order does not have longer meaning.

https://bugs.php.net/bug.php?id=63816

I have a question if could be possible to change that interface order to not cause this error or change the minimal required PHP version for PRADO 4.2.0?

Thanks in advance for answer.

Best regards,
Marcin Haba (gani)

**Comment by ctrlaltca** [comment]
Hello Marcin, nice catch!
Thank you for the detailed report, I've committed a fix in https://github.com/pradosoft/prado/commit/ae7e69a044822c2cdaceb3033903912ac146edd4
I'll let it settle for a few days and eventually release 4.2.1.
Let me know if it works for you, thank you

**Comment by ganiuszka** [comment]
Hello ctrlaltca,

Thank you for the quick fix.

I tested it. Yes, I confirm that it works well with PHP 7.3.

Thanks again!

Best regards,
Marcin Haba (gani)

**Comment by ctrlaltca** [comment]
Fixed in ae7e69a044822c2cdaceb3033903912ac146edd4
